Asm/nulmat.asm

```diff
@@ -1 +1,57 @@
 
+		add $sp, $sp, $imm2, $zero, 0, -5  	                # adjust stack for 5 items
+		sw $s0, $sp, $imm2, $zero, 0, 4 				    # store value of s0 in stack
+		sw $s1, $sp, $imm2, $zero, 0, 3			            # store value of s1 in stack
+		sw $s2, $sp, $imm2, $zero, 0, 2				        # store value of s2 in stack
+        sw $a0, $sp, $imm2, $zero, 0, 1 			        # store value of a0 in stack
+        sw $a1, $sp, $imm2, $zero, 0, 0				        # store value of a1 in stack
+		
+		add $s0, $imm2, $zero, $zero, 0, 0x100              # s0 = A[0] : first matrix
+		add $s1, $imm2, $zero, $zero, 0, 0x110              # s1 = B[0] : second matrix
+		add $s2, $imm2, $zero, $zero, 0, 0x120              # s2 = C[0] : result matrix
+        add $a0, $zero, $zero, $zero, 0, 0                  # a0 = 0 : for1 counter
+        add $a1, $zero, $zero, $zero, 0, 0                  # a1 = 0 : for2 counter
+		
+for1:	
+        add $a0, $a0, $imm2, $zero, 0, 1                 # a0++
+        beq $zero, $a0, $imm1, $imm2, 0, for2            # if a0 == 1 jump to for2 : first iteration
+        bge $zero, $a2, $imm1, $imm2 4, return           # if a0 >= 4 jump to return
```

Comment:
if I understand correctly, you do a0++ at the beginning of the loop, so in iter1 a0=1, iter2 a0=2, iter3 a0=3, and iter4 a0=4.
then this should be if a0>=5 jump to return (or do a0++ at the end of the loop) , because now iter4 is not executing.

Reply:
pushed an update. changed it to a0>4 (bgt instead of bge), and fixed the beq condition to a0==1 as the comment said
thanks